sdk/tables/data-tables/src/TableServiceClient.ts

```diff
@@ -156,6 +156,7 @@ export class TableServiceClient {
       | TableServiceClientOptions,
     options?: TableServiceClientOptions
   ) {
+    options = { ... options };
```

Comment:
I think this fix only works if you pass in a credential, but the problem persists if you pass options as the second argument.

Perhaps a better fix would be to remove this line `clientOptions.endpoint = clientOptions.endpoint || this.url;` and instead setting the endpoint using a fallback within `internalPipelineOptions` like:

```ts
const internalPipelineOptions: InternalClientPipelineOptions = {
      ...clientOptions,
      endpoint: clientOptions.endpoint ?? this.url
```